mosqlient/registry/predictions.py

```diff
@@ -0,0 +1,178 @@
+from pydantic import BaseModel, ConfigDict
+
+import requests
+import nest_asyncio
+import pandas as pd
+
+from mosqlient import types
+from mosqlient.client import Client
+from mosqlient.requests import get_all
+from mosqlient.errors import ClientError, PredictionPostError
+from mosqlient.config import API_DEV_URL, API_PROD_URL
+
+nest_asyncio.apply()
+
+
+def _params(**kwargs) -> dict[str, Any]:
+    params = {}
+    for k, v in kwargs.items():
+        if isinstance(v, (bool, int, float, str)):
+            params[k] = str(v)
+        elif v is None:
+            continue
+        else:
+            raise TypeError(f"Unknown type f{type(v)}")
+
+    return params
+
+
+class Prediction(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    client: Client | None
+
+    @classmethod
+    def get(cls, **kwargs):
+        """
+        [DOCUMENTATION](https://api.mosqlimate.org/docs/registry/GET/predictions/)
+
+        SEARCH PARAMETERS
+        All parameters are Optional, use them to filter the Predictions in the
+        result
+
+        id [int]: Search by Prediction ID
+        model_id [int]: Search by Model ID
+        model_name [str]: Search by Model Name
+        model_ADM_level [int]: Search by ADM Level
+        model_time_resolution [str]: Search by Time Resolution
+        model_disease [str]: Search by Disease
+        author_name [str]: Search by Author Name
+        author_username [str]: Search by Author Username
+        author_institution [str]: Search by Author Institution
+        repository [str]: Search by GitHub repository
+        implementation_language [str]: Search by Implementation Language
+        temporal [bool]: Search by Temporal Models
+        spatial [bool]: Search by Spatial Models
+        categorical [bool]: Search by Categorical Models
+        commit [str]: Search by Git Commit
+        predict_date [str]: Search by prediction date. Format: YYYY-MM-DD
+        start [str]: Search by prediction start date. Format: YYYY-MM-DD
+        end [str]: Search by prediction end date. Format: YYYY-MM-DD
+        """
+        env = kwargs["env"] if "env" in kwargs else "prod"
+        timeout = kwargs["timeout"] if "timeout" in kwargs else 60
+
+        PredictionGETParams(**kwargs)
+        params = _params(**kwargs)
+
+        async def fetch_prediction():
+            return await get_all(
+                "registry",
+                "predictions",
+                params,
+                env=env,
+                timeout=timeout
+            )
+
+        if asyncio.get_event_loop().is_running():
+            loop = asyncio.get_event_loop()
+            future = asyncio.ensure_future(fetch_prediction())
+            return loop.run_until_complete(future)
+        return asyncio.run(fetch_prediction())
+
+    def post(
+        self,
+        model_id: int,
+        description: str,
+        commit: str,
+        predict_date: str,
+        prediction: pd.DataFrame,
```

Comment:
@fccoelho do you think its fine if a DataFrame is needed to upload a prediction rather than a dict/json?